app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -163,11 +158,28 @@ define(function (require, exports, module) {
     },
 
     deleteDisplayedAccountProfileImage (account) {
-      return account.deleteAvatar(this._displayedProfileImage.get('id'))
-        .then(() => {
-          // A blank image will clear the cache
-          this.updateProfileImage(new ProfileImage(), account);
-        });
+      if ( ! account.get('profileImageId') ) {
+        return account.fetchCurrentProfileImage()
```

Comment:
This looks like the proper solution. I'm going to propose a slight simplification that uses promise chaining that eliminates duplicate code but still Does The Right Thing.

```js
return p().then(() => {
  if (! account.get('profileImageId')) {
    return account.fetchCurrentProfileImage()
      .then((profileImage) => {
        // Cache the result to make sure we don't flash the default
        // image while fetching the latest profile image
        // this should set the profileImageId on the account.
        this._updateCachedProfileImage(profileImage, account);
      });
  }
  // if we reach here, the account has a profile image ID already.
})
// by this point, the account will have an avatar and a profileImageId
.then(() => account.deleteAvatar(account.get('profileImageId')))
.then(() => this.updateProfileImage(new ProfileImage(), account));
```

Do you have much experience with promises? If not, they are a bit strange to get used to, but once you do, they are pretty great. A good intro is: https://developers.google.com/web/fundamentals/getting-started/primers/promises